File: ng4/src/app/app.component.ts
import { Component } from '@angular/core';
import { AuthService } from './auth.service';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  providers: [AuthService]
})
export class AppComponent {
  title = 'app';

  constructor(public auth: AuthService) {
    auth.handleAuthentication();
  }
}

File: dartweb/dartang/lib/app_component.dart
import 'package:angular/angular.dart';
import 'package:angular_components/angular_components.dart';
import 'src/evaporating_cloud/evaporating_cloud_component.dart';

// AngularDart info: https://webdev.dartlang.org/angular
// Components info: https://webdev.dartlang.org/components

@Component(
  selector: 'my-app',
  providers: const [
    materialProviders],
  styleUrls: const [
    'package:angular_components/app_layout/layout.scss.css',
    'app_component.css'],
  templateUrl: 'app_component.html',
  directives: [
    EvaporatingCloudComponent,
    DeferredContentDirective,
    MaterialButtonComponent,
    MaterialIconComponent,
    MaterialPersistentDrawerDirective,
    MaterialToggleComponent,
    MaterialListComponent,
    MaterialListItemComponent,],
)
class AppComponent {

}
